crypto/x509/x509_vfy.c

```diff
@@ -3153,17 +3164,12 @@ static int build_chain(X509_STORE_CTX *ctx)
                         search = 0;
                         continue;
                     }
-                    ss = cert_self_signed(ctx, x);
-                    if (ss < 0) {
-                        X509err(X509_F_BUILD_CHAIN, ERR_R_INTERNAL_ERROR);
-                        ctx->error = X509_V_ERR_UNSPECIFIED;
-                        return 0;
-                    }
+                    self_signed = cert_self_signed(ctx, x, 0) == 1;
```

Comment:
This change looks wrong to me.  It fails to deal with unrecoverable errors from `cert_self_signed` (see line 2988).

Reply:
Good point.
When improving the commit structure yesterday I was already wondering about this change and the similar one mentioned next.
I cannot recall why I did that; likely its was due to a glitch when changing forth and back parts of `build_chain()`.

So I've re-added the return on error, yet while deliberately leaving out
```
X509err(X509_F_BUILD_CHAIN, ERR_R_INTERNAL_ERROR);
```
because the function called just before: `cert_self_signed(ctx, cert, 0)`
is responsible for providing a suitable error queue entry, and meanwhile it actually does so in all error situations.